### PR TITLE
feat: add support to NOT LIKE operator

### PR DESCRIPTION
**Pull request by dacopan**
### SUMMARY
Change `get_sqla_query` method of ` helpers.py`  to support `NOT_LIKE` operator

### TESTING INSTRUCTIONS
All tests should pass

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/helpers.py [hunk 1]
```diff
@@ -2112,21 +2120,22 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
...
         filter_columns = [flt.get("col") for flt in filter] if filter else []
         rejected_filter_columns = [
-            col
-            for col in filter_columns
-            if col
-            and not is_adhoc_column(col)
-            and col not in self.column_names
-            and col not in applied_template_filters
-        ] + rejected_adhoc_filters_columns
+                                      col
```
Comment (mistercrunch) [inline superset/models/helpers.py:2123]: looks like our linters are behaving different, follow the steps here and you should get our standard dev tools like ruff and pylint through the pip install, and pre-commit hooks setup

https://superset.apache.org/docs/contributing/development/#installing-development-tools

File: superset/models/helpers.py [hunk 2]
```diff
@@ -1873,9 +1874,9 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                     if (
                         op
                         not in {
-                            utils.FilterOperator.EQUALS.value,
```
Comment (mistercrunch) [inline superset/models/helpers.py:1876]: I'm confused as this looks properly indented in `master` RN...    
https://github.com/apache/superset/blob/master/superset/models/helpers.py#L1875-L1878 (?)

Comment (dacopan) [reply superset/models/helpers.py]: I updated using ruff and formated correctly

File: superset/models/helpers.py (hunk 1 above)
Comment (dacopan) [reply superset/models/helpers.py]: I updated using ruff and formated correctly

**Comment by rusackas** [comment]
Running CI 🤞

**Comment by dacopan** [comment]
I don't understand why fail front end test, my changes not affect to failed tests

**Comment by rusackas** [comment]
Re-running that one... it might just be flaky, I've seen this one fail randomly before.

**Comment by dacopan** [comment]
I no have option to re-run

**Comment by dacopan** [comment]
@rusackas  please your help with the pipeline, in my local server all test run OK, I don't understand why fail in test that files that not changed from master

**Comment by rusackas** [comment]
Running it again... ping me if it flakes out :)

**Review by mistercrunch — APPROVED** [review]
LGTM

**Pull request merged**